sdks/python/apache_beam/dataframe/frames_test.py

```diff
@@ -263,6 +263,39 @@ def test_where_concrete_args(self):
             }), axis=1),
         df)
 
+  def test_combine_dataframe(self):
+    with expressions.allow_non_parallel_operations():
+      df = pd.DataFrame({'A': [0, 0], 'B': [4, 4]})
+      df2 = pd.DataFrame({'A': [1, 1], 'B': [3, 3]})
+      take_smaller = lambda s1, s2: s1 if s1.sum() < s2.sum() else s2
+      self._run_test(lambda df, df2: df.combine(df2, take_smaller), df, df2)
```

Comment:
```suggestion
    df = pd.DataFrame({'A': [0, 0], 'B': [4, 4]})
    df2 = pd.DataFrame({'A': [1, 1], 'B': [3, 3]})
    take_smaller = lambda s1, s2: s1 if s1.sum() < s2.sum() else s2
    self._run_test(lambda df, df2: df.combine(df2, take_smaller), df, df2, nonparallel=True)
```

I recently added this option, `nonparallel` which makes _run_test do some additional verification (including checking that a `reason` was specified). Please use that instead of the allow_non_parallel_operations block